src/main/java/eu/xlime/kafka/rdf/BaseXLiMeResourceToMongo.java

```diff
@@ -69,30 +69,36 @@ public BaseXLiMeResourceToMongo(Properties props) {
 	}
 
 	@Override
-	protected String generateSummary() {
-		return String.format("%s_%s processed=%s %s", 
-				getClass().getSimpleName(), instId, getMessagesProcessed(), summariseResourcesProcessed());
-		
+	protected String generateSummaryString() {
+		Summary s = generateSummary();
+		return s.toString();
 	}
 	
-	private String summariseResourcesProcessed() {
+	protected Summary generateSummary(){
+		Summary sum = new Summary();
+		sum.setConsumerId(getClass().getSimpleName() + "_" + instId);
+		sum.setCounter("messagesProcessed", getMessagesProcessed());
+		sum.addCounters(summariseResourcesProcessed());
+		return sum;
+	}
+
+	private Map<String,Long> summariseResourcesProcessed() {
 		Set<Class<? extends XLiMeResource>> clzz = ImmutableSet.<Class<? extends XLiMeResource>>builder()
 				.addAll(resourcesRead.keySet()).addAll(resourcesStored.keySet()).build();
-		if (clzz.isEmpty()) return "\n\tNo xLiMe resources read or stored yet";
-		StringBuilder sb = new StringBuilder();
+		if (clzz.isEmpty()) return null;

```

Comment:
Better to return an empty Map, otherwise calling this function we have to check for a null value.